src/Pipes/Kafka.hs

```diff
@@ -1,104 +1,104 @@
-kafkaSink co partition = PS.bracket connect release publish
+  :: (MonadIO m, MonadSafe m)
+  => ProducerProperties
+  -> TopicName
+  -> Consumer BS.ByteString m ()
+kafkaSink props topic = PS.bracket connect release publish
   where
-    connect = liftIO $ do
-      kafka <- HK.newKafka HK.RdKafkaProducer (co ^. kafkaOverrides)
-      HK.addBrokers kafka (co ^. hostString)
-      topic <- HK.newKafkaTopic kafka (co ^. topicName) (co ^. topicOverrides)
-      return (kafka, topic)
-    release (fst -> k) = liftIO $ HK.drainOutQueue k
-    publish (snd -> t) = forever $ do
-      m <- P.await
-      liftIO $ HK.produceMessage t partition $ KafkaProduceMessage m
+    connect = do
+      res <- liftIO $ newProducer props
+      case res of
+        Left err       -> liftIO $ throwM err
+        Right producer -> pure producer
+    release producer = liftIO $ closeProducer producer
+    publish producer =
+      forever $ do
+        m <- P.await
+        liftIO $ produceMessage producer $ mkMessage topic Nothing (Just m)
+    mkMessage t k v =
+      ProducerRecord
+      {prTopic = t, prPartition = UnassignedPartition, prKey = k, prValue = v}
 
 --------------------------------------------------------------------------------
 kafkaSource
-  :: forall m. (MonadSafe m)
-  => ConnectOpts
-  -> Int
-  -> KafkaOffset
-  -> MessageTimeout
-  -> Producer KafkaMessage m ()
-kafkaSource co partition offset to = PS.bracket connect release consume
+  :: (MonadIO m, MonadSafe m)
+  => ConsumerProperties
+  -> Subscription
+  -> Timeout
+  -> Producer (ConsumerRecord (Maybe BS.ByteString) (Maybe BS.ByteString)) m ()
+kafkaSource props subs timeout@(Timeout timeoutms) =
+  PS.bracket connect release consume
   where
-    connect = liftIO $ do
-      kafka <- HK.newKafka HK.RdKafkaConsumer (co ^. kafkaOverrides)
-      HK.addBrokers kafka (co ^. hostString)
-      topic <- HK.newKafkaTopic kafka (co ^. topicName) (co ^. topicOverrides)
-      HK.startConsuming topic partition offset
-      return (kafka, topic)
-    release (_kafka, topic) = liftIO $ HK.stopConsuming topic partition
-    consume :: (Kafka, KafkaTopic) -> Producer KafkaMessage m ()
-    consume (k, t) = do
-      ret <- liftIO $ HK.consumeMessage t partition (timeoutToInt to)
-      case ret of
-        Left _err -> return ()
-        Right m -> do
-          P.yield m
-          consume (k ,t)
+    connect = do
+      res <- liftIO $ newConsumer props subs
+      case res of
+        Left err       -> liftIO $ throwM err
+        Right consumer -> pure consumer
+    release consumer = liftIO $ closeConsumer consumer
+    consume consumer = do
+      res <- liftIO $ pollMessage consumer timeout
+      case res of
+        Left (KafkaResponseError RdKafkaRespErrPartitionEof) ->
+          waitForMessages consumer timeoutms
+        Left (KafkaResponseError RdKafkaRespErrTimedOut) ->
+          waitForMessages consumer 0
+        Left err -> do
+          liftIO $ print err
```

Comment:
I suppose we might as well fix thise liftIO prints with a proper logging library too..